internal/clients/team4/client.go

```diff
@@ -200,3 +202,64 @@ func (c *client) VoteForElection(roleToElect shared.Role, candidateList []shared
 
 	return returnList
 }
+
+
+//MonitorIIGORole decides whether to perform monitoring on a role
+//COMPULOSRY: must be implemented
+func (c *client) MonitorIIGORole(roleName shared.Role) bool {
+	
+	presidentID := c.getPresident()
+	speakerID := c.getSpeaker()
+	judgeID := c.getJudge()
+	clientID := shared.ClientID(4)
+	// TODO: Choose sensible thresholds!
+	trustThreshold := 0.5
+	resourcesThreshold := shared.Resources(100)
+	monitoring := false
+	if presidentID == clientID {
+		// If we are the president.
+		monitoring = (c.internalParam.agentsTrust[speakerID] < trustThreshold ||
+					  c.internalParam.agentsTrust[judgeID]   < trustThreshold ) && 
+					 (c.ServerReadHandle.GetGameState().ClientInfo.Resources > resourcesThreshold)		   
+	} else if speakerID == clientID {
+		// If we are the Speaker.
+		monitoring = (c.internalParam.agentsTrust[presidentID] < trustThreshold ||
+					  c.internalParam.agentsTrust[judgeID]   < trustThreshold ) && 
+					 (c.ServerReadHandle.GetGameState().ClientInfo.Resources > resourcesThreshold)
+	} else if judgeID == clientID {
+		// If we are the Judge.
+		monitoring = (c.internalParam.agentsTrust[speakerID] < trustThreshold ||
+					  c.internalParam.agentsTrust[judgeID]   < trustThreshold ) && 
+					 (c.ServerReadHandle.GetGameState().ClientInfo.Resources > resourcesThreshold)
```

Comment:
Minor suggestion, but wouldn't a switch statement be more readable here?

Reply:
Yes agreed, I personally am not a fan of switch statements but sure, I am gonna change it.